### PR TITLE
enh(a11y): New user modal

### PR DESCRIPTION
**Pull request by Pytal**
- For https://github.com/nextcloud/server/issues/37091
- Supersede https://github.com/nextcloud/server/pull/35898

## Summary

| | Screenshot |
--- | ---
**Before** | image
**After** | image

Before | After
--- | ---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]
Fancy.
Not tested, and probably need a double check as there is an eslint error.

**Review by szaimen — APPROVED** [review]
Tested and seems to work but didnt review the code. 

Looks much better now! :)

**Review by nimishavijay — DISMISSED** [review]
Nice! Great improvement :) Some points regarding the "One of either password or email..." message, no other feedback, looks great :)

- [ ] We could show it as a subline under both the password and email fields to avoid confusion? Then the "(required)" next to the labels can also be removed. This way if the form is submitted without either of those fields filled, it's easy to indicate the error as the field already has the sublines for info. What do you think? cc @nextcloud/designers
- [ ] If we are not doing that, the message should be left aligned with the rest of the labels
- [ ] Would the wording "Either password or email is required" be better? @jancborchardt and @Pytal

**Review by marcoambrosini — APPROVED** [review]

**Comment by szaimen** [comment]
@Pytal drone failure apparently related

**Comment by Pytal** [comment]
> Nice! Great improvement :) Some points regarding the "One of either password or email..." message, no other feedback, looks great :)
> 
> * [ ]  We could show it as a subline under both the password and email fields to avoid confusion? Then the "(required)" next to the labels can also be removed. This way if the form is submitted without either of those fields filled, it's easy to indicate the error as the field already has the sublines for info. What do you think? cc @nextcloud/designers

Duplicating the text as sublines seems strange 🤔

For invalid input the native form validation should suffice
image

> * [x]  If we are not doing that, the message should be left aligned with the rest of the labels
> * [x]  Would the wording "Either password or email is required" be better? @jancborchardt and @Pytal

**Review by JuliaKirschenheuter — APPROVED** [review]
Tested ;) Works, code looks good

File: apps/settings/src/components/Users/NewUserModal.vue [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+	computed: {
+		settings() {
+			return this.$store.getters.getServerData
+		},
+
+		usernameLabel() {
+			if (this.settings.newUserGenerateUserID) {
+				return t('settings', 'Username will be autogenerated')
+			}
+			return t('settings', 'Username (required)')
```
Comment (szaimen) [inline apps/settings/src/components/Users/NewUserModal.vue:238]: Small suggestion (not sure if we use that somewhere but would possibly make sense to make clear that it cannot be changed afterwards?):

```suggestion
			return t('settings', 'User id (required)')
```

File: apps/settings/src/components/Users/NewUserModal.vue [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+				:label-visible="true"
+				autocapitalize="none"
+				autocomplete="off"
+				autocorrect="off"
+				pattern="[a-zA-Z0-9 _\.@\-']+"
+				required />
+			<NcTextField class="modal__item"
+				data-test="displayName"
+				:value.sync="newUser.displayName"
+				:label="t('settings', 'Display name')"
```
Comment (szaimen) [inline apps/settings/src/components/Users/NewUserModal.vue:47]: Maybe?

```suggestion
				:label="t('settings', 'Display name of user')"
```

File: apps/settings/src/components/Users/NewUserModal.vue (hunk 1 above)
Comment (Pytal) [reply apps/settings/src/components/Users/NewUserModal.vue:238]: @nimishavijay?

File: apps/settings/src/components/Users/NewUserModal.vue (hunk 2 above)
Comment (Pytal) [reply apps/settings/src/components/Users/NewUserModal.vue:47]: @nimishavijay?

**Comment by Pytal** [comment]
What do you think about @szaimen's wording suggestions for follow-up @jancborchardt?

**Pull request merged**